File: backend/app/schemas/watchlist.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class WatchlistEntryBase(BaseModel):
    account_number: Optional[str] = Field(None, description="Exact account number")
    ifsc_code: Optional[str] = Field(None, description="IFSC code (often paired with account)")
    upi_id: Optional[str] = Field(None, description="UPI ID")
    phone: Optional[str] = Field(None, description="Phone number")
    reason: str = Field(..., description="Reason for watchlisting")
    risk_score: float = Field(85.0, description="Override risk score, usually 85 or 100")
    is_active: bool = Field(True, description="Active status")

class WatchlistEntryCreate(WatchlistEntryBase):
    pass

class WatchlistEntryUpdate(BaseModel):
    reason: Optional[str] = None
    risk_score: Optional[float] = None
    is_active: Optional[bool] = None
    phone: Optional[str] = None

class WatchlistEntryOut(WatchlistEntryBase):
    id: str
    added_by_user_id: Optional[str]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/watchlist.ts
import { api } from './client';

export interface WatchlistEntry {
  id: string;
  account_number: string | null;
  ifsc_code: string | null;
  upi_id: string | null;
  phone: string | null;
  reason: string;
  risk_score: number;
  is_active: boolean;
  added_by_user_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface WatchlistEntryCreate {
  account_number: string | null;
  ifsc_code: string | null;
  upi_id: string | null;
  phone: string | null;
  reason: string;
  risk_score: number;
  is_active: boolean;
}

export interface WatchlistEntryUpdate {
  reason?: string;
  risk_score?: number;
  is_active?: boolean;
  phone?: string;
}

export const watchlistApi = {
  list: async (): Promise<WatchlistEntry[]> => {
    const res = await api.get('/watchlist/');
    return res.data;
  },

  create: async (data: WatchlistEntryCreate): Promise<WatchlistEntry> => {
    const res = await api.post('/watchlist/', data);
    return res.data;
  },

  update: async (id: string, data: WatchlistEntryUpdate): Promise<WatchlistEntry> => {
    const res = await api.put(`/watchlist/${id}`, data);
    return res.data;
  },

  delete: async (id: string): Promise<void> => {
    await api.delete(`/watchlist/${id}`);
  }
};
